### ISSUE TITLE
Expose not working

### ISSUE THREAD

**Issue by if-kenn** [ailispaw/docker-root#6]
I don't know if this really is an issue of docker-root or not.

I am running the following container:

```
docker run                          \
  --restart=always                  \
  --detach                          \
  -e MYSQL_ALLOW_EMPTY_PASSWORD=yes \
  --publish=3306:3306               \
  --name mariadb10-0                \
  mariadb:10.0
```

From within the docker VM, sometimes I can use the IP of the host to connect to MariaDB (MySQL) and sometimes I can't. Any thoughts?

**Comment by if-kenn** [comment]
I just confirmed that on a CentOS 7.1 docker host with the above, I can connect to the MariaDB using the docker0 (172.17.42.1) or eth0 (10.0.0.197) IP address of the host.

**Comment by if-kenn** [comment]
The strange thing the same thing is not true for port 80 or 8080.

**Comment by ailispaw** [comment]
> sometimes

Hmm...

**Comment by if-kenn** [comment]
I think I might blow away my whole docker-root-xhyve as I am wondering if the port somehow got held onto as restoring a database took all the space on the vm so I had to resize drive and reboot.

**Comment by ailispaw** [comment]
How's `sudo iptables -L` in the host VM?
My case;

```
[docker@docker-root ~]$ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (1 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             172.17.0.1           tcp dpt:mysql
```

**Comment by if-kenn** [comment]
I am in the process up upgrading brew and re-cloning docker-root-xhyve. If I have the issue again, I will report the iptables.

**Comment by ailispaw** [comment]
Thanks. And I am about to upgrade DockerRoot as well.
It has debug mode for Docker daemon.

**Comment by if-kenn** [comment]
I can now access the published port. 

For now I think it is a legacy issue from running out of space with a MySQL DB restore and the port not being expose properly. I will reopen this issue if I see it again.

**Comment by ailispaw** [comment]
:)

**Comment by ailispaw** [comment]
I just got an iptable error with MySQL container on start as blow.

```
ERRO[0000] Error response from daemon: Cannot start container db: iptables failed: iptables --wait -t filter -A DOCKER ! -i docker0 -o docker0 -p tcp -d 172.17.0.3 --dport 3306 -j ACCEPT:  (wait: no child processes)
```

It was the first time for me and just once. I will keep it in mind and reopen it.

**Comment by if-kenn** [comment]
I just got this as well:

```
docker start nginx1-6
Error response from daemon: Cannot start container nginx1-6: iptables failed: iptables --wait -t nat -A DOCKER -p tcp -d 0/0 --dport 8080 -j DNAT --to-destination 172.17.0.6:80 ! -i docker0:  (wait: no child processes)
```

I waited a short bit and tried again and it worked.

**Comment by if-kenn** [comment]
But it is now looking like it is not publishing port 80.

**Comment by if-kenn** [comment]
Looks like something might be going haywire with route. I get this:

```
telnet 192.168.64.2 8080
telnet: can't connect to remote host (192.168.64.2): No route to host
```

and 

```
telnet 172.17.42.1 8080
telnet: can't connect to remote host (172.17.42.1): No route to host
```

This pauses for a bit for the end lines:

```
[docker@docker-root ~]$ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (1 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             172.17.0.4           tcp dpt:mysql
ACCEPT     tcp  --  anywhere             172.17.0.8           tcp dpt:www
```

**Comment by ailispaw** [comment]
Hmm.. Is that telnet from your Mac?

**Comment by if-kenn** [comment]
No the telnet is from the VM which was really weird to me.

**Comment by ailispaw** [comment]
Could you try it with debug mode?

**Comment by ailispaw** [comment]
Wait a minute. Could you use DockerRoot v1.0.8?

**Comment by if-kenn** [comment]
Sure, what do I need to put it into debug mode?

**Comment by if-kenn** [comment]
Sure, will shut this one down and stash the dir.

**Comment by if-kenn** [comment]
FYI, thought I would check before doing that. After doing a "make halt" and restarting my Mac, everything is working on current specs.

Will now degrade to 1.0.8

**Comment by ailispaw** [comment]
OK. Thanks for the report.

**Comment by if-kenn** [comment]
FYI, when issuing a "make halt" I see this:

```
spawn_id: spawn id exp7 not open
    while executing
"interact"
```

**Comment by ailispaw** [comment]
It is from makefile on your Mac, not from VM. It is harmless.

**Comment by if-kenn** [comment]
OK, I have a LEMP stack running 4 different sites on DockerRoot 1.0.8, have held off on the 5 one since the DB made me grow the HD image. I will work with these for a while and see I run into any routing problems. Let me know if there is anything in particular you want me to do/check.

**Comment by ailispaw** [comment]
I have made a custom Docker v1.8.3 binary by myself since v1.0.9.
I guess it may cause that problem.
Please use v1.0.8, which has the original version from Docker, for a while to see how it is good or not.

**Comment by ailispaw** [comment]
I found a bug in my modification of Docker.
I will release v1.1.0 soon.

**Comment by ailispaw** [comment]
Just released DockerRoot v1.1.0 and updated docker-root-xhyve, too.

@if-kenn Thank you so much for your support.
